diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index 0cae2df..b910c4f 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -39,7 +39,7 @@ android {
     }
 
     signingConfigs {
-        register("debug") {
+        named("debug") {
             storeFile = rootProject.file("keystore/debug.keystore")
             storePassword = "android"
             keyAlias = "androiddebugkey"
