diff --git a/app/src/main/kotlin/org/fossify/gallery/activities/SettingsActivity.kt b/app/src/main/kotlin/org/fossify/gallery/activities/SettingsActivity.kt
index 80c51f0..77cf16d 100644
--- a/app/src/main/kotlin/org/fossify/gallery/activities/SettingsActivity.kt
+++ b/app/src/main/kotlin/org/fossify/gallery/activities/SettingsActivity.kt
@@ -77,6 +77,7 @@ class SettingsActivity : SimpleActivity() {
         setupHideSystemUI()
         setupHiddenItemPasswordProtection()
         setupExcludedItemPasswordProtection()
+        setupIncludedItemPasswordProtection()
         setupAppPasswordProtection()
         setupFileDeletionPasswordProtection()
         setupDeleteEmptyFolders()
@@ -211,7 +212,9 @@ class SettingsActivity : SimpleActivity() {
             if (isRPlus() && !isExternalStorageManager()) {
                 GrantAllFilesDialog(this)
             } else {
-                startActivity(Intent(this, IncludedFoldersActivity::class.java))
+                handleIncludedFolderPasswordProtection {
+                    startActivity(Intent(this, IncludedFoldersActivity::class.java))
+                }
             }
         }
     }
@@ -434,6 +437,28 @@ class SettingsActivity : SimpleActivity() {
         }
     }
 
+    private fun setupIncludedItemPasswordProtection() {
+        binding.settingsIncludedItemPasswordProtection.isChecked = config.isIncludedPasswordProtectionOn
+        binding.settingsIncludedItemPasswordProtectionHolder.setOnClickListener {
+            val tabToShow = if (config.isIncludedPasswordProtectionOn) config.includedProtectionType else SHOW_ALL_TABS
+            SecurityDialog(this, config.includedPasswordHash, tabToShow) { hash, type, success ->
+                if (success) {
+                    val hasPasswordProtection = config.isIncludedPasswordProtectionOn
+                    binding.settingsIncludedItemPasswordProtection.isChecked = !hasPasswordProtection
+                    config.isIncludedPasswordProtectionOn = !hasPasswordProtection
+                    config.includedPasswordHash = if (hasPasswordProtection) "" else hash
+                    config.includedProtectionType = type
+
+                    if (config.isIncludedPasswordProtectionOn) {
+                        val confirmationTextId = if (config.includedProtectionType == PROTECTION_FINGERPRINT)
+                            org.fossify.commons.R.string.fingerprint_setup_successfully else org.fossify.commons.R.string.protection_setup_successfully
+                        ConfirmationDialog(this, "", confirmationTextId, org.fossify.commons.R.string.ok, 0) { }
+                    }
+                }
+            }
+        }
+    }
+
     private fun setupAppPasswordProtection() {
         binding.settingsAppPasswordProtection.isChecked = config.isAppPasswordProtectionOn
         binding.settingsAppPasswordProtectionHolder.setOnClickListener {
diff --git a/app/src/main/kotlin/org/fossify/gallery/extensions/Activity.kt b/app/src/main/kotlin/org/fossify/gallery/extensions/Activity.kt
index bc9e03b..6667071 100644
--- a/app/src/main/kotlin/org/fossify/gallery/extensions/Activity.kt
+++ b/app/src/main/kotlin/org/fossify/gallery/extensions/Activity.kt
@@ -959,6 +959,18 @@ fun Activity.handleExcludedFolderPasswordProtection(callback: () -> Unit) {
     }
 }
 
+fun Activity.handleIncludedFolderPasswordProtection(callback: () -> Unit) {
+    if (config.isIncludedPasswordProtectionOn) {
+        SecurityDialog(this, config.includedPasswordHash, config.includedProtectionType) { _, _, success ->
+            if (success) {
+                callback()
+            }
+        }
+    } else {
+        callback()
+    }
+}
+
 fun Activity.openRecycleBin() {
     Intent(this, MediaActivity::class.java).apply {
         putExtra(DIRECTORY, RECYCLE_BIN)
diff --git a/app/src/main/kotlin/org/fossify/gallery/helpers/Config.kt b/app/src/main/kotlin/org/fossify/gallery/helpers/Config.kt
index 87723e0..a81d474 100644
--- a/app/src/main/kotlin/org/fossify/gallery/helpers/Config.kt
+++ b/app/src/main/kotlin/org/fossify/gallery/helpers/Config.kt
@@ -139,6 +139,18 @@ class Config(context: Context) : BaseConfig(context) {
         get() = prefs.getInt(EXCLUDED_PROTECTION_TYPE, PROTECTION_PATTERN)
         set(excludedProtectionType) = prefs.edit().putInt(EXCLUDED_PROTECTION_TYPE, excludedProtectionType).apply()
 
+    var isIncludedPasswordProtectionOn: Boolean
+        get() = prefs.getBoolean(INCLUDED_PASSWORD_PROTECTION, false)
+        set(isIncludedPasswordProtectionOn) = prefs.edit().putBoolean(INCLUDED_PASSWORD_PROTECTION, isIncludedPasswordProtectionOn).apply()
+
+    var includedPasswordHash: String
+        get() = prefs.getString(INCLUDED_PASSWORD_HASH, "")!!
+        set(includedPasswordHash) = prefs.edit().putString(INCLUDED_PASSWORD_HASH, includedPasswordHash).apply()
+
+    var includedProtectionType: Int
+        get() = prefs.getInt(INCLUDED_PROTECTION_TYPE, PROTECTION_PATTERN)
+        set(includedProtectionType) = prefs.edit().putInt(INCLUDED_PROTECTION_TYPE, includedProtectionType).apply()
+
     fun addIncludedFolder(path: String) {
         val currIncludedFolders = HashSet<String>(includedFolders)
         currIncludedFolders.add(path)
diff --git a/app/src/main/kotlin/org/fossify/gallery/helpers/Constants.kt b/app/src/main/kotlin/org/fossify/gallery/helpers/Constants.kt
index 1d70cab..fca404a 100644
--- a/app/src/main/kotlin/org/fossify/gallery/helpers/Constants.kt
+++ b/app/src/main/kotlin/org/fossify/gallery/helpers/Constants.kt
@@ -12,6 +12,9 @@ const val TEMPORARILY_SHOW_EXCLUDED = "temporarily_show_excluded"
 const val EXCLUDED_PASSWORD_PROTECTION = "excluded_password_protection"
 const val EXCLUDED_PASSWORD_HASH = "excluded_password_hash"
 const val EXCLUDED_PROTECTION_TYPE = "excluded_protection_type"
+const val INCLUDED_PASSWORD_PROTECTION = "included_password_protection"
+const val INCLUDED_PASSWORD_HASH = "included_password_hash"
+const val INCLUDED_PROTECTION_TYPE = "included_protection_type"
 const val IS_THIRD_PARTY_INTENT = "is_third_party_intent"
 const val AUTOPLAY_VIDEOS = "autoplay_videos"
 const val REMEMBER_LAST_VIDEO_POSITION = "remember_last_video_position"
diff --git a/app/src/main/res/layout/activity_settings.xml b/app/src/main/res/layout/activity_settings.xml
index ace6c9e..5111f54 100644
--- a/app/src/main/res/layout/activity_settings.xml
+++ b/app/src/main/res/layout/activity_settings.xml
@@ -807,6 +807,21 @@
 
             </RelativeLayout>
 
+            <RelativeLayout
+                android:id="@+id/settings_included_item_password_protection_holder"
+                style="@style/SettingsHolderSwitchStyle"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content">
+
+                <org.fossify.commons.views.MyMaterialSwitch
+                    android:id="@+id/settings_included_item_password_protection"
+                    style="@style/SettingsSwitchStyle"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:text="@string/password_protect_included" />
+
+            </RelativeLayout>
+
             <RelativeLayout
                 android:id="@+id/settings_file_deletion_password_protection_holder"
                 style="@style/SettingsHolderSwitchStyle"
diff --git a/app/src/main/res/values/strings.xml b/app/src/main/res/values/strings.xml
index b3b424b..2935461 100644
--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -232,6 +232,7 @@
     <string name="paths_imported_successfully">Paths imported successfully</string>
     <string name="media_management_prompt">To make sure that all file operations work reliably, please make this app a Media management app in your device settings.</string>
     <string name="password_protect_excluded">Password protect excluded folder visibility</string>
+    <string name="password_protect_included">Password protect included folder visibility</string>
     <string name="media_management_manual">Something went wrong, please go into your device Settings - Apps - Special app access - Media management apps and allow this app to manage media.</string>
     <string name="media_management_note">If the redirection does not work, please go into your device Settings - Apps - Special app access - Media management apps and allow this app to manage media.</string>
     <string name="media_management_alternative">If you do not want to do it, you can also go into your device Settings - Apps - Special app access - Media management apps and allow this app to manage media.</string>
