diff --git a/app/src/main/kotlin/org/fossify/gallery/activities/MainActivity.kt b/app/src/main/kotlin/org/fossify/gallery/activities/MainActivity.kt
index 47f356b..779d313 100644
--- a/app/src/main/kotlin/org/fossify/gallery/activities/MainActivity.kt
+++ b/app/src/main/kotlin/org/fossify/gallery/activities/MainActivity.kt
@@ -463,9 +463,6 @@ class MainActivity : SimpleActivity(), DirectoryOperationsListener {
             findItem(R.id.temporarily_show_hidden).isVisible = !config.shouldShowHidden
             findItem(R.id.stop_showing_hidden).isVisible =
                 (!isRPlus() || isExternalStorageManager()) && config.temporarilyShowHidden
-
-            findItem(R.id.temporarily_show_excluded).isVisible = !config.temporarilyShowExcluded
-            findItem(R.id.stop_showing_excluded).isVisible = config.temporarilyShowExcluded
         }
     }
 
@@ -502,8 +499,6 @@ class MainActivity : SimpleActivity(), DirectoryOperationsListener {
                 R.id.change_view_type -> changeViewType()
                 R.id.temporarily_show_hidden -> tryToggleTemporarilyShowHidden()
                 R.id.stop_showing_hidden -> tryToggleTemporarilyShowHidden()
-                R.id.temporarily_show_excluded -> tryToggleTemporarilyShowExcluded()
-                R.id.stop_showing_excluded -> tryToggleTemporarilyShowExcluded()
                 R.id.create_new_folder -> createNewFolder()
                 R.id.open_recycle_bin -> openRecycleBin()
                 R.id.column_count -> changeColumnCount()
@@ -718,24 +713,6 @@ class MainActivity : SimpleActivity(), DirectoryOperationsListener {
         refreshMenuItems()
     }
 
-    private fun tryToggleTemporarilyShowExcluded() {
-        if (config.temporarilyShowExcluded) {
-            toggleTemporarilyShowExcluded(false)
-        } else {
-            handleExcludedFolderPasswordProtection {
-                toggleTemporarilyShowExcluded(true)
-            }
-        }
-    }
-
-    private fun toggleTemporarilyShowExcluded(show: Boolean) {
-        mLoadedInitialPhotos = false
-        config.temporarilyShowExcluded = show
-        binding.directoriesGrid.adapter = null
-        getDirectories()
-        refreshMenuItems()
-    }
-
     override fun deleteFolders(folders: ArrayList<File>) {
         val fileDirItems = folders
             .asSequence()
diff --git a/app/src/main/res/menu/menu_main.xml b/app/src/main/res/menu/menu_main.xml
index 193d0e1..ab34354 100644
--- a/app/src/main/res/menu/menu_main.xml
+++ b/app/src/main/res/menu/menu_main.xml
@@ -38,16 +38,6 @@
         android:showAsAction="never"
         android:title="@string/stop_showing_hidden"
         app:showAsAction="never" />
-    <item
-        android:id="@+id/temporarily_show_excluded"
-        android:showAsAction="never"
-        android:title="@string/temporarily_show_excluded"
-        app:showAsAction="never" />
-    <item
-        android:id="@+id/stop_showing_excluded"
-        android:showAsAction="never"
-        android:title="@string/stop_showing_excluded"
-        app:showAsAction="never" />
     <item
         android:id="@+id/set_as_default_folder"
         android:showAsAction="never"
diff --git a/app/src/main/res/menu/menu_main_intent.xml b/app/src/main/res/menu/menu_main_intent.xml
index 75e5a61..bf9d4f3 100644
--- a/app/src/main/res/menu/menu_main_intent.xml
+++ b/app/src/main/res/menu/menu_main_intent.xml
@@ -13,14 +13,4 @@
         android:showAsAction="never"
         android:title="@string/stop_showing_hidden"
         app:showAsAction="never" />
-    <item
-        android:id="@+id/temporarily_show_excluded"
-        android:showAsAction="never"
-        android:title="@string/temporarily_show_excluded"
-        app:showAsAction="never" />
-    <item
-        android:id="@+id/stop_showing_excluded"
-        android:showAsAction="never"
-        android:title="@string/stop_showing_excluded"
-        app:showAsAction="never" />
 </menu>
