diff --git a/app/src/main/kotlin/org/fossify/gallery/activities/MainActivity.kt b/app/src/main/kotlin/org/fossify/gallery/activities/MainActivity.kt
index 779d313..e43ab79 100644
--- a/app/src/main/kotlin/org/fossify/gallery/activities/MainActivity.kt
+++ b/app/src/main/kotlin/org/fossify/gallery/activities/MainActivity.kt
@@ -98,7 +98,6 @@ import org.fossify.gallery.extensions.getFavoritePaths
 import org.fossify.gallery.extensions.getNoMediaFoldersSync
 import org.fossify.gallery.extensions.getOTGFolderChildrenNames
 import org.fossify.gallery.extensions.getSortedDirectories
-import org.fossify.gallery.extensions.handleExcludedFolderPasswordProtection
 import org.fossify.gallery.extensions.handleMediaManagementPrompt
 import org.fossify.gallery.extensions.isDownloadsFolder
 import org.fossify.gallery.extensions.launchAbout
diff --git a/app/src/main/kotlin/org/fossify/gallery/activities/SettingsActivity.kt b/app/src/main/kotlin/org/fossify/gallery/activities/SettingsActivity.kt
index 77cf16d..7c43c7a 100644
--- a/app/src/main/kotlin/org/fossify/gallery/activities/SettingsActivity.kt
+++ b/app/src/main/kotlin/org/fossify/gallery/activities/SettingsActivity.kt
@@ -76,7 +76,6 @@ class SettingsActivity : SimpleActivity() {
         setupScreenRotation()
         setupHideSystemUI()
         setupHiddenItemPasswordProtection()
-        setupExcludedItemPasswordProtection()
         setupIncludedItemPasswordProtection()
         setupAppPasswordProtection()
         setupFileDeletionPasswordProtection()
@@ -221,7 +220,7 @@ class SettingsActivity : SimpleActivity() {
 
     private fun setupManageExcludedFolders() {
         binding.settingsManageExcludedFoldersHolder.setOnClickListener {
-            handleExcludedFolderPasswordProtection {
+            handleIncludedFolderPasswordProtection {
                 startActivity(Intent(this, ExcludedFoldersActivity::class.java))
             }
         }
@@ -414,29 +413,6 @@ class SettingsActivity : SimpleActivity() {
         }
     }
 
-    private fun setupExcludedItemPasswordProtection() {
-        binding.settingsExcludedItemPasswordProtectionHolder.beGoneIf(binding.settingsHiddenItemPasswordProtectionHolder.isVisible())
-        binding.settingsExcludedItemPasswordProtection.isChecked = config.isExcludedPasswordProtectionOn
-        binding.settingsExcludedItemPasswordProtectionHolder.setOnClickListener {
-            val tabToShow = if (config.isExcludedPasswordProtectionOn) config.excludedProtectionType else SHOW_ALL_TABS
-            SecurityDialog(this, config.excludedPasswordHash, tabToShow) { hash, type, success ->
-                if (success) {
-                    val hasPasswordProtection = config.isExcludedPasswordProtectionOn
-                    binding.settingsExcludedItemPasswordProtection.isChecked = !hasPasswordProtection
-                    config.isExcludedPasswordProtectionOn = !hasPasswordProtection
-                    config.excludedPasswordHash = if (hasPasswordProtection) "" else hash
-                    config.excludedProtectionType = type
-
-                    if (config.isExcludedPasswordProtectionOn) {
-                        val confirmationTextId = if (config.excludedProtectionType == PROTECTION_FINGERPRINT)
-                            org.fossify.commons.R.string.fingerprint_setup_successfully else org.fossify.commons.R.string.protection_setup_successfully
-                        ConfirmationDialog(this, "", confirmationTextId, org.fossify.commons.R.string.ok, 0) { }
-                    }
-                }
-            }
-        }
-    }
-
     private fun setupIncludedItemPasswordProtection() {
         binding.settingsIncludedItemPasswordProtection.isChecked = config.isIncludedPasswordProtectionOn
         binding.settingsIncludedItemPasswordProtectionHolder.setOnClickListener {
diff --git a/app/src/main/res/layout/activity_settings.xml b/app/src/main/res/layout/activity_settings.xml
index 5111f54..bccdb77 100644
--- a/app/src/main/res/layout/activity_settings.xml
+++ b/app/src/main/res/layout/activity_settings.xml
@@ -792,21 +792,6 @@
 
             </RelativeLayout>
 
-            <RelativeLayout
-                android:id="@+id/settings_excluded_item_password_protection_holder"
-                style="@style/SettingsHolderSwitchStyle"
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content">
-
-                <org.fossify.commons.views.MyMaterialSwitch
-                    android:id="@+id/settings_excluded_item_password_protection"
-                    style="@style/SettingsSwitchStyle"
-                    android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
-                    android:text="@string/password_protect_excluded" />
-
-            </RelativeLayout>
-
             <RelativeLayout
                 android:id="@+id/settings_included_item_password_protection_holder"
                 style="@style/SettingsHolderSwitchStyle"
diff --git a/app/src/main/res/values-de/strings.xml b/app/src/main/res/values-de/strings.xml
index 0511b0c..6c26d00 100644
--- a/app/src/main/res/values-de/strings.xml
+++ b/app/src/main/res/values-de/strings.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <string name="app_launcher_name">Galerie</string>
+    <string name="app_launcher_name">Galerie Kids</string>
     <string name="edit">Bearbeiten</string>
     <string name="open_camera">Kamera öffnen</string>
     <string name="hidden">(versteckt)</string>
diff --git a/app/src/main/res/values/strings.xml b/app/src/main/res/values/strings.xml
index 2c0a867..4ff34c5 100644
--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -232,7 +232,7 @@
     <string name="paths_imported_successfully">Paths imported successfully</string>
     <string name="media_management_prompt">To make sure that all file operations work reliably, please make this app a Media management app in your device settings.</string>
     <string name="password_protect_excluded">Password protect excluded folder visibility</string>
-    <string name="password_protect_included">Password protect included folder visibility</string>
+    <string name="password_protect_included">Password protect included &amp; excluded folder visibility</string>
     <string name="media_management_manual">Something went wrong, please go into your device Settings - Apps - Special app access - Media management apps and allow this app to manage media.</string>
     <string name="media_management_note">If the redirection does not work, please go into your device Settings - Apps - Special app access - Media management apps and allow this app to manage media.</string>
     <string name="media_management_alternative">If you do not want to do it, you can also go into your device Settings - Apps - Special app access - Media management apps and allow this app to manage media.</string>
